app/controllers/user.js

```diff
@@ -18,4 +37,41 @@ userController.addUserWithFacebook = function(request, response) {
     });
 };
 
+userController.sendPasswordResetEmail = function(userEmail) {
+    crypto.randomBytes(20, function(err, buffer) {
```

Comment:
Not a show stopper, but it would be good to eventually move this into a static method on the PasswordReset model, something like `createResetToken()` or similar.

Then you could do:
```
PasswordReset.createResetToken()
  .then(function(token) {
    // create and save the new password reset doc
  })